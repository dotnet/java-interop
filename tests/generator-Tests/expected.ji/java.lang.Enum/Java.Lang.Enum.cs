using System;
using System.Collections.Generic;
using Android.Runtime;
using Java.Interop;

namespace Java.Lang {

	// Metadata.xml XPath class reference: path="/api/package[@name='java.lang']/class[@name='Enum']"
	[global::Android.Runtime.Register ("java/lang/Enum", DoNotGenerateAcw=true)]
	[global::Java.Interop.JavaTypeParameters (new string [] {"E extends java.lang.Enum<E>"})]
	public abstract partial class Enum : global::Java.Lang.Object, global::Java.Lang.IComparable {
		static readonly JniPeerMembers _members = new JniPeerMembers ("java/lang/Enum", typeof (Enum));

		internal static new IntPtr class_ref {
			get { return _members.JniPeerType.PeerReference.Handle; }
		}

		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
		public override global::Java.Interop.JniPeerMembers JniPeerMembers {
			get { return _members; }
		}

		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
		protected override IntPtr ThresholdClass {
			get { return _members.JniPeerType.PeerReference.Handle; }
		}

		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
		protected override global::System.Type ThresholdType {
			get { return _members.ManagedPeerType; }
		}

		protected Enum (IntPtr javaReference, JniHandleOwnership transfer) : base (javaReference, transfer)
		{
		}

		// Metadata.xml XPath method reference: path="/api/package[@name='java.lang']/class[@name='Enum']/method[@name='compareTo' and count(parameter)=1 and parameter[1][@type='E']]"
		[Register ("compareTo", "(Ljava/lang/Enum;)I", "")]
		public unsafe int CompareTo (global::Java.Lang.Object o)
		{
			const string __id = "compareTo.(Ljava/lang/Enum;)I";
			IntPtr native_o = JNIEnv.ToLocalJniHandle (o);
			try {
				JniArgumentValue* __args = stackalloc JniArgumentValue [1];
				__args [0] = new JniArgumentValue (native_o);
				var __rm = _members.InstanceMethods.InvokeNonvirtualInt32Method (__id, this, __args);
				return __rm;
			} finally {
				JNIEnv.DeleteLocalRef (native_o);
				global::System.GC.KeepAlive (o);
			}
		}

	}

	[global::Android.Runtime.Register ("java/lang/Enum", DoNotGenerateAcw=true)]
	internal partial class EnumInvoker : Enum, global::Java.Lang.IComparable {
		public EnumInvoker (IntPtr handle, JniHandleOwnership transfer) : base (handle, transfer)
		{
		}

		static readonly JniPeerMembers _members = new JniPeerMembers ("java/lang/Enum", typeof (EnumInvoker));

		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
		public override global::Java.Interop.JniPeerMembers JniPeerMembers {
			get { return _members; }
		}

		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
		protected override global::System.Type ThresholdType {
			get { return _members.ManagedPeerType; }
		}

	}
}
