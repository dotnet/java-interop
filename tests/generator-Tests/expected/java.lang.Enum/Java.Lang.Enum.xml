<?xml version="1.0" encoding="UTF-8" ?>
<api>
	<package name="java.lang">
		<class abstract="false" deprecated="not deprecated" final="false" name="Object" static="false" visibility="public">
		</class>
		<interface abstract="true" deprecated="not deprecated" final="false" name="Comparable" static="false" visibility="public">
			<typeParameters>
				<typeParameter name="T" />
			</typeParameters>
			<method abstract="true" deprecated="not deprecated" final="false" name="compareTo" native="false" return="int" static="false" synchronized="false" visibility="public">
				<parameter name="another" type="T">
				</parameter>
			</method>
		</interface>
		<class abstract="true" deprecated="not deprecated" extends="java.lang.Object" extends-generic-aware="java.lang.Object" final="false" name="Enum" static="false" visibility="public">
		<typeParameters>
		<typeParameter name="E">
			<genericConstraints>
				<genericConstraint type="java.lang.Enum&lt;E&gt;" />
			</genericConstraints>
		</typeParameter>
		</typeParameters>
		<implements name="java.lang.Comparable" name-generic-aware="java.lang.Comparable&lt;E&gt;">
		</implements>
		<method abstract="false" deprecated="not deprecated" final="true" name="compareTo" native="false" return="int" static="false" synchronized="false" visibility="public">
			<parameter name="o" type="E">
			</parameter>
		</method>
		</class>
		<class abstract="false" deprecated="not deprecated" extends="java.lang.Enum" extends-generic-aware="java.lang.Enum&lt;java.lang.State>" final="true" name="State" static="true" visibility="public">
			<field deprecated="not deprecated" final="true" name="BLOCKED" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
			<field deprecated="not deprecated" final="true" name="NEW" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
			<field deprecated="not deprecated" final="true" name="RUNNABLE" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
			<field deprecated="not deprecated" final="true" name="TERMINATED" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
			<field deprecated="not deprecated" final="true" name="TIMED_WAITING" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
			<field deprecated="not deprecated" final="true" name="WAITING" static="true" transient="false" type="java.lang.State" type-generic-aware="java.lang.State" visibility="public" volatile="false">
			</field>
		</class>
	</package>
</api>

